### ISSUE TITLE
out of memory when run on testnet

### ISSUE THREAD

**Issue by jerry6290** [Bytom/bytom#593]
#### Environment

OS: Ubuntu 16.04.3 LTS (GNU/Linux 4.4.0-109-generic x86_64)
go version 1.10.1

#### Problem Description
out of memory，Exception Stack
```go
time="2018-04-11T13:57:45Z" level=info msg="BlockResponseMessage height:29"
time="2018-04-11T13:57:46Z" level=error msg="Authenticate fail" error="Serve: no token"
time="2018-04-11T13:57:46Z" level=error msg="Authenticate fail" error="Serve: no token"
time="2018-04-11T13:57:48Z" level=info msg="Synchronising ..."
time="2018-04-11T13:57:48Z" level=error msg="Authenticate fail" error="Serve: no token"
time="2018-04-11T13:57:48Z" level=error msg="Authenticate fail" error="Serve: no token"
time="2018-04-11T13:57:50Z" level=error msg="Authenticate fail" error="Serve: no token"
time="2018-04-11T13:57:50Z" level=error msg="Authenticate fail" error="Serve: no token"
fatal error: runtime: out of memory

runtime stack:
runtime.throw(0xcd3887, 0x16)
	/usr/local/go/src/runtime/panic.go:616 +0x81
runtime.sysMap(0xc44e940000, 0x8000000, 0x0, 0x1527ed8)
	/usr/local/go/src/runtime/mem_linux.go:216 +0x20a
runtime.(*mheap).sysAlloc(0x150e7a0, 0x8000000, 0xa81333)
	/usr/local/go/src/runtime/malloc.go:470 +0xd4
runtime.(*mheap).grow(0x150e7a0, 0x4000, 0x0)
	/usr/local/go/src/runtime/mheap.go:907 +0x60
runtime.(*mheap).allocSpanLocked(0x150e7a0, 0x4000, 0x1527ee8, 0x1506270)
	/usr/local/go/src/runtime/mheap.go:820 +0x301
runtime.(*mheap).alloc_m(0x150e7a0, 0x4000, 0x450101, 0x99e500)
	/usr/local/go/src/runtime/mheap.go:686 +0x118
runtime.(*mheap).alloc.func1()
	/usr/local/go/src/runtime/mheap.go:753 +0x4d
runtime.(*mheap).alloc(0x150e7a0, 0x4000, 0xc421010101, 0x7ffd5c653d10)
	/usr/local/go/src/runtime/mheap.go:752 +0x8a
runtime.largeAlloc(0x8000000, 0xc421d10101, 0x1)
	/usr/local/go/src/runtime/malloc.go:826 +0x94
runtime.mallocgc.func1()
	/usr/local/go/src/runtime/malloc.go:721 +0x46
runtime.systemstack(0x0)
	/usr/local/go/src/runtime/asm_amd64.s:409 +0x79
runtime.mstart()
	/usr/local/go/src/runtime/proc.go:1175

goroutine 45 [running]:
runtime.systemstack_switch()
	/usr/local/go/src/runtime/asm_amd64.s:363 fp=0xc42095d988 sp=0xc42095d980 pc=0x459410
runtime.mallocgc(0x8000000, 0xb881c0, 0x29484bc9f7adf001, 0xc44920a000)
	/usr/local/go/src/runtime/malloc.go:720 +0x8a2 fp=0xc42095da28 sp=0xc42095d988 pc=0x4128c2
runtime.makeslice(0xb881c0, 0x1000000, 0x1000000, 0xc44920a000, 0x400000, 0x400000)
	/usr/local/go/src/runtime/slice.go:61 +0x77 fp=0xc42095da58 sp=0xc42095da28 pc=0x443687
github.com/bytom/mining/tensority.mulMatrix(0xc4207a9920, 0x20, 0x20, 0xc428c44000, 0x400000, 0x470000, 0x20, 0x1d, 0x20000000000ccccc)
	/usr/local/go/bin/src/github.com/bytom/mining/tensority/matrix.go:33 +0x2ff fp=0xc42095dbe8 sp=0xc42095da58 pc=0xa4f87f
github.com/bytom/mining/tensority.Hash(0xc42095dcb8, 0xc420252288, 0x5c2c62fba503ebaf)
	/usr/local/go/bin/src/github.com/bytom/mining/tensority/tensority.go:10 +0xf9 fp=0xc42095dc58 sp=0xc42095dbe8 pc=0xa50999
github.com/bytom/consensus/difficulty.CheckProofOfWork(0xc42095dcb8, 0xc420252288, 0x20000000000ccccc, 0xf949edeff4e4dcb8)
	/usr/local/go/bin/src/github.com/bytom/consensus/difficulty/difficulty.go:121 +0x35 fp=0xc42095dc88 sp=0xc42095dc58 pc=0xa51045
github.com/bytom/mining/cpuminer.(*CPUMiner).solveBlock(0xc420125180, 0xc421d17680, 0xc420094870, 0xc420088720, 0x0)
	/usr/local/go/bin/src/github.com/bytom/mining/cpuminer/cpuminer.go:70 +0x1d6 fp=0xc42095ddb0 sp=0xc42095dc88 pc=0xa810e6
github.com/bytom/mining/cpuminer.(*CPUMiner).generateBlocks(0xc420125180, 0xc420088720)
	/usr/local/go/bin/src/github.com/bytom/mining/cpuminer/cpuminer.go:102 +0x183 fp=0xc42095dfd0 sp=0xc42095ddb0 pc=0xa81333
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:2361 +0x1 fp=0xc42095dfd8 sp=0xc42095dfd0 pc=0x45bfa1
created by github.com/bytom/mining/cpuminer.(*CPUMiner).miningWorkerController.func1
	/usr/local/go/bin/src/github.com/bytom/mining/cpuminer/cpuminer.go:136 +0x80
```
-----

**Comment by shengling2008** [comment]
git branch 看下分支名字，git log 看下commit

**Comment by HAOYUatHZ** [comment]
@jerry6290 which version u r running

**Comment by HAOYUatHZ** [comment]
@jerry6290 r u running it on a vps

**Comment by HAOYUatHZ** [comment]
r u running with "--mining" option

**Comment by HAOYUatHZ** [comment]
email me (haoyu#protonmail.com) ur wechat id, i would like to have further discussion with u.

**Comment by jerry6290** [comment]
- version:0.4.6
- OS: Ubuntu 16.04.3 LTS (GNU/Linux 4.4.0-109-generic x86_64)
- go version 1.10.1
- CPU: 1 vCore
- RAM: 1024 MB

step:
1. git clone src code, build with 
```shell
$ make bytomd    # build bytomd
$ make bytomcli  # build bytomcli
```
2. Initialize
```shell
$ cd ./cmd/bytomd
$ ./bytomd init --chain_id testnet
```

3. luach
```shell
$ ./bytomd node --mining
```

4. run successful,but about some minutes, the mining node stop with the out of memory error。
    If restart the node, more block and tx will by syn,but it would stop by the same error.

**Comment by Blockmeta** [comment]
fixed at version 1.0.2
